### PR TITLE
[stdlib] Remove `theGlobalMT19937`

### PR DESCRIPTION
**Pull request by benrimmington**
SwiftPrivate/PRNG.swift:

- currently uses `theGlobalMT19937`;
- previously used `arc4random` (see #1939);
- is obsoleted by SE-0202: Random Unification.

### CODE REVIEW

File: stdlib/private/SwiftPrivate/ShardedAtomicCounter.swift [hunk 1]
```diff
@@ -67,7 +67,7 @@ public struct _stdlib_ShardedAtomicCounter {
     var _state: Int
 
     public init() {
-      _state = Int(Int32(bitPattern: rand32()))
+      _state = Int.random(in: .min ... .max)
```
Comment (benrimmington) [inline stdlib/private/SwiftPrivate/ShardedAtomicCounter.swift]: Should this be `_state = Int(Int32.random(in: .min ... .max))` instead?

File: validation-test/stdlib/Set.swift [hunk 2]
```diff
@@ -1141,10 +1128,21 @@ SetTestSuite.test("deleteChainCollision2") {
...
+  _ c1: C1, _ c2: C2
+) -> [(C1.Iterator.Element, C2.Iterator.Element)] {
+  var result: [(C1.Iterator.Element, C2.Iterator.Element)] = []
+  for e1 in c1 {
+    for e2 in c2 {
+      result.append((e1, e2))
+    }
+  }
+  return result
+}
```
Comment (benrimmington) [inline validation-test/stdlib/Set.swift]: This is copied from [validation-test/stdlib/Dictionary.swift](https://github.com/apple/swift/blob/8c6df3647b3eec0ee526051e412bebb2c83b29d0/validation-test/stdlib/Dictionary.swift#L1505-L1519)

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by benrimmington** [comment]
@lorentey Thanks for your review.

The "deleteChainCollisionRandomized" tests were printing the `srandom` seed, which won't have been useful since [December 2015](https://github.com/apple/swift/commit/3961b71e4e8a509b26e9e1db5294ff6c08510e5f).

I've copied a linear congruential generator from the `RandomShuffle` benchmark. Printing its seed directly to stdout only seems to work in `--verbose` mode, so instead I've added a message argument to each `expectNotEqual` and `expectNotNil` call.

I've used a random `collisionChains` and `chainOverlap`, instead of the `cartesianProduct(_:_:)` in both tests. This makes them run faster locally (4 minutes instead of 12 minutes) but I suspect it won't make such a difference for swift-ci.

File: validation-test/stdlib/Set.swift [hunk 3]
```diff
@@ -1175,16 +1157,16 @@ SetTestSuite.test("deleteChainCollisionRandomized") {
...
 
   var s = Set<TestKeyTy>(minimumCapacity: 30)
   for i in 1..<300 {
-    let key = getKey(uniformRandom(collisionChains * chainLength))
-    if uniformRandom(chainLength * 2) == 0 {
+    let key = getKey(Int.random(in: 0 ..< (collisionChains * chainLength), using: &generator))
+    if Int.random(in: 0 ..< (chainLength * 2), using: &generator) == 0 {
       s.remove(key)
     } else {
       s.insert(TestKeyTy(key.value))
```
Comment (benrimmington) [inline validation-test/stdlib/Set.swift]: @lorentey Is this correct?

* `SetTestSuite` is creating a new `TestKeyTy` for each insertion (with a matching `value` and `hashValue`).
* `DictionaryTestSuite` reuses the `TestKeyTy` from `getKey(_:)` (with a separately randomized `hashValue`).

**Comment by benrimmington** [comment]
@lorentey Please take another look at this if you can.

I've added the `LinearCongruentialGenerator` since your last review.

**Comment by lorentey** [comment]
Sorry about the delay!

@swift-ci test

File: validation-test/stdlib/Set.swift (hunk 3 above)
Comment (lorentey) [reply validation-test/stdlib/Set.swift]: It sure doesn't look correct. I believe the `Set` tests should also insert the key from `getKey(_:)`. I'll review & fix in a followup.

**Review by lorentey — APPROVED** [review]

**Pull request merged**